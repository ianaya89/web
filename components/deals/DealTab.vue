<template>
  <div class="inactive-tab bg-white w-full px-2 py-3 flex border-b border-r border-grey-l4" :class="getTabClass" @click="$emit('clicked', deal.id)">
    <div class="w-12 mr-4">
      <slot name="logo" :deal="deal">
        <avatar class="avatar-active mt-2" :classes="'w-12 h-12'" :src="$get(deal, 'brand.logo.url')" />
      </slot>
    </div>
    <div class="w-3/4 flex flex-col">
      <p class="text-h7 lg:text-menu font-medium font-sans leading-tight pt-2">
        <slot name="name" :deal="deal">
          {{ $get(deal, 'brand.name', '-') }}
        </slot>
      </p>
      <p class="text-h7 lg:text-p3 uppercase leading-tight font-sans" :class="getStatusColor">
        {{ $t('app.deals.status.'+deal.status) }}
      </p>
      <p class="text-h7 lg:text-menu text-grey font-medium font-sans leading-tight mt-2">
        <span v-if="deal.priceTo==null || deal.priceTo==0">{{ $get(deal, 'price') | currency }}</span>
        <span v-else>{{ $get(deal, 'price') | currency }} - {{ $get(deal, 'priceTo') | currency }}</span>
      </p>
    </div>
  </div>
</template>

<script>
import Avatar from '@/components/Avatar'
import deal from '@/services/mixins/deal'

export default {
  name: 'DealTab',

  components: {
    Avatar
  },

  mixins: [deal],

  props: {
    deal: {
      type: Object,
      required: true
    },

    active: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    getTabClass() {
      return this.active ? 'active-tab' : 'cursor-pointer'
    },

    getStatusColor() {
      return 'text-' + this.getDealStatusColor(this.deal) + '-d1'
    }
  }
}
</script>

<style lang="postcss" scoped>
.inactive-tab {
  @apply .border-l-2 .border-transparent;
  border-bottom-color: #f2f2f2;
  border-right-color: #f2f2f2;
}

.active-tab {
  @apply .border-l-2 .border-red .bg-grey-l4;
  border-bottom-color: #f2f2f2;
  border-right-color: #f2f2f2;

  .avatar-active {
    div {
      @apply .border .border-grey-l2;
    }
  }
}
</style>
