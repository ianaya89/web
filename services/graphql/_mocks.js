export const mockArray = (count = 1, props = ['id']) => {
  return [...new Array(count)].map((_, k) => {
    const mockedObject = {}

    props.forEach(p => {
      mockedObject[p] = `mocked ${p} ${k}`
    })

    return mockedObject
  })
}

export const mockObjects = (count = 1, cb = () => ({ id: 1 })) => {
  return [...new Array(count)].map((_, k) => {
    return cb(k)
  })
}
