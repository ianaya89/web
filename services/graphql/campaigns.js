import gql from 'graphql-tag'

export const CampaignsQuery = {
  mock: null,
  query: gql`
    query Campaigns($id: ID!) {
      unit(id: $id) {
        id
        entity {
          id
          ... on Team {
            campaigns {
              id
              name
              message
              openRate
              sendAt
              processedAt
              inventoryItems {
                id
              }
              packageItems {
                id
              }
              recipients {
                id
                company
              }
            }
          }
          ... on League {
            campaigns {
              id
              name
              message
              openRate
              sendAt
              processedAt
              inventoryItems {
                id
              }
              packageItems {
                id
              }
              recipients {
                id
                company
              }
            }
          }
        }
      }
    }
  `
}

export const CampaignsPaginatedQuery = {
  mock: null,
  query: gql`
    query Campaigns($id: ID!, $page: Int = 1, $size: Int!) {
      unit(id: $id) {
        id
        entity {
          id
          ... on Team {
            campaignsPaginator(count: $size, page: $page) {
              data {
                id
                name
                message
                openRate
                sendAt
                processedAt
                inventoryItems {
                  id
                }
                packageItems {
                  id
                }
                recipients {
                  id
                  company
                }
              }
              paginatorInfo {
                lastPage
                total
              }
            }
          }
          ... on League {
            campaignsPaginator(count: $size, page: $page) {
              data {
                id
                name
                message
                openRate
                sendAt
                processedAt
                inventoryItems {
                  id
                }
                packageItems {
                  id
                }
                recipients {
                  id
                  company
                }
              }
              paginatorInfo {
                lastPage
                total
              }
            }
          }
        }
      }
    }
  `
}

export const CreateCampaignMutation = {
  mutation: gql`
    mutation CreateCampaign($input: CreateCampaignInput!) {
      createCampaign(input: $input) {
        id
        name
      }
    }
  `
}
