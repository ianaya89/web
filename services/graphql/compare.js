import gql from 'graphql-tag'
import { mapValues, omit } from 'lodash'
import {
  currencyFormatter,
  dateFormatter,
  numberFormatter,
  percentageFormatter
} from '../../plugins/filters'
import { normalizeCategoryName, get } from '../helpers'

export const CompareItemsQuery = {
  mock: [],
  query: gql`
    query CompareItems(
      $ids: [ID!]
      $season: SeasonAttr!
      $seasonString: String!
    ) {
      compareMarketplaceItems(ids: $ids) {
        id
        slots {
          id
          slot
          quantity
        }
        price
        category {
          id
          name
          target
          path
        }
        unit {
          id
          entity {
            id
            name
            logo {
              url
            }
            ... on Team {
              teamLeagues(season: $season) {
                league {
                  id
                  name
                  logo {
                    url
                  }
                }
              }
              groupedSocialAccounts {
                type
                followersCount
              }
              socialAccounts {
                id
                name
                followersCount
                type
                precalcs(periodicity: SEASONAL, season: $seasonString) {
                  data {
                    totalEngagement
                  }
                }
              }
              precalcs(periodicity: SEASONAL, season: $seasonString) {
                data {
                  reach {
                    totalAttendance
                    uniqueWebsiteVisits
                    socialMedia
                    broadcasting
                    printOutdoor
                    emailsSent
                  }
                  fans {
                    seasonTicket
                    members
                    newsletter
                    crm
                    mobileApp
                  }
                  genderSplit {
                    male
                    female
                    unknown
                  }
                  ageSplit {
                    group17
                    group18
                    group25
                    group35
                    group45
                    group55
                    group65
                  }
                  matchVisitors {
                    avgStadiumAttendance
                  }
                }
              }
              city {
                id
                name
              }
              country {
                id
                name
              }
              profile {
                founded
              }
              teamStadiums(season: { get: SPECIFIC, value: $seasonString }) {
                stadium {
                  id
                  precalcs(periodicity: SEASONAL, season: $seasonString) {
                    data {
                      capacity
                      visitorsPerMatch
                      visitorsTotal
                      businessSeats
                      skyboxes
                    }
                  }
                }
                metaData {
                  visitorsPerMatch
                  matchesCount
                  visitorsTotal
                }
              }
            }
            ... on League {
              groupedSocialAccounts {
                type
                followersCount
              }
              socialAccounts {
                id
                type
                precalcs(periodicity: SEASONAL, season: $seasonString) {
                  data {
                    totalEngagement
                  }
                }
              }
              precalcs(periodicity: SEASONAL, season: $seasonString) {
                data {
                  reach {
                    totalAttendance
                    uniqueWebsiteVisits
                    socialMedia
                    broadcasting
                    printOutdoor
                    emailsSent
                  }
                  fans {
                    seasonTicket
                    members
                    newsletter
                    website
                    socialMedia
                  }
                  genderSplit {
                    male
                    female
                    unknown
                  }
                  ageSplit {
                    group17
                    group18
                    group25
                    group35
                    group45
                    group55
                    group65
                  }
                }
              }
              country {
                id
                name
              }
              profile {
                founded
              }
            }
          }
        }
      }
    }
  `,
  fetchPolicy: 'network-only',
  update(data) {
    data.compareMarketplaceItems = get(data, 'compareMarketplaceItems', []).map(
      item => {
        const stadiumData = {
          capacity: get(
            item,
            'unit.entity.teamStadiums.0.stadium.precalcs.0.data.capacity'
          ),
          visitorsPerMatch: get(
            item,
            'unit.entity.teamStadiums.0.metaData.visitorsPerMatch'
          ),
          visitorsTotal: get(
            item,
            'unit.entity.teamStadiums.0.metaData.visitorsTotal'
          ),
          businessSeats: get(
            item,
            'unit.entity.teamStadiums.0.stadium.precalcs.0.data.businessSeats'
          ),
          skyboxes: get(
            item,
            'unit.entity.teamStadiums.0.stadium.precalcs.0.data.skyboxes'
          )
        }

        return {
          inventoryName: [
            item.category.target === 'package'
              ? 'package.title'
              : normalizeCategoryName(item.category.path.split(' / ')[0]),
            normalizeCategoryName(
              item.category.name,
              item.category.target === 'package'
            )
          ],
          price: currencyFormatter(item.price),
          availableTime: (item.slots || []).map(s => s.slot).join(', '),
          basic: {
            id: item.id,
            logo: item.unit.entity.logo.url,
            title: item.unit.entity.name,
            unit_id: item.unit.id,
            visits: item.unit.entity.precalcs[0].data.reach.uniqueWebsiteVisits,
            info: ''
          },
          sections: {
            general: {
              founded: dateFormatter(item.unit.entity.profile.founded),
              location: `${get(item, 'unit.entity.city.name', '')}, ${get(
                item,
                'unit.entity.country.name',
                ''
              )}`,
              stadiumCapacity: numberFormatter(
                get(
                  item,
                  'unit.entity.teamStadiums.0.stadium.precalcs.0.data.capacity',
                  null
                )
              ),
              averageAttendance: numberFormatter(
                get(
                  item,
                  'unit.entity.teamStadiums.0.metaData.visitorsPerMatch',
                  null
                )
              )
            },
            gender: omit(
              mapValues(
                get(item, 'unit.entity.precalcs.0.data.genderSplit', {}),
                reachItemValue => percentageFormatter(reachItemValue)
              ),
              ['__typename']
            ),
            age: omit(
              mapValues(
                get(item, 'unit.entity.precalcs.0.data.ageSplit', {}),
                reachItemValue => percentageFormatter(reachItemValue)
              ),
              ['__typename']
            ),
            socialmedia: omit(get(item, 'unit.entity.socialAccounts'), [
              '__typename'
            ]),
            fans: omit(
              mapValues(
                get(item, 'unit.entity.precalcs.0.data.fans', {
                  total: null,
                  seasonTicket: null,
                  members: null,
                  newsletter: null,
                  crm: null,
                  mobileApp: null
                }),
                fansItemValue => numberFormatter(fansItemValue)
              ),
              ['__typename']
            ),
            stadium: omit(
              mapValues(stadiumData, stadiumItemValue => {
                if (typeof stadiumItemValue === 'boolean') {
                  return {
                    translate: true,
                    value: stadiumItemValue
                  }
                } else {
                  return numberFormatter(stadiumItemValue)
                }
              }),
              ['__typename']
            ),
            league: {
              leagueName: get(
                item,
                'unit.entity.teamLeagues.0.league.name',
                '-'
              )
            }
          }
        }
      }
    )

    return data.compareMarketplaceItems
  }
}
