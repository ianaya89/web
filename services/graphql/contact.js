import gql from 'graphql-tag'

// TODO: Endpoint required
export const ContactsQuery = {
  mock: [],
  query: gql`
    query Contacts($id: ID!) {
      unit(id: $id) {
        id
        contacts {
          id
          name
          company
          email
        }
      }
    }
  `,
  update(data) {
    return data.unit.contacts
  }
}

export const SubmitContactMutation = {
  mutation: gql`
    mutation SubmitContact($input: CreateContactInput!) {
      createContact(input: $input) {
        id
        name
        company
        email
      }
    }
  `
}
