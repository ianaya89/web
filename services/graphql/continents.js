import gql from 'graphql-tag'

export const ContinentsQuery = {
  mock: [],
  query: gql`
    query Continents {
      continents {
        id
        name
      }
    }
  `,
  result: data => {
    data.data.continents.sort((a, b) => (a.name > b.name ? 1 : -1))
    return data
  }
}

export const ContinentById = {
  mock: [],
  query: gql`
    query Continents($id: ID!) {
      continentById(id: $id) {
        id
        name
        countries {
          id
          name
        }
      }
    }
  `
}
