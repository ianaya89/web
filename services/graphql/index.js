export * from './login'
export * from './seasons'
export * from './forgot'
export * from './reset'
export * from './user'
export * from './blog'
export * from './unit'
export * from './tips'
export * from './history'
export * from './cups'
export * from './fans'
export * from './reach'
export * from './exposures'
export * from './payments'
export * from './plans'
export * from './subscriptions'
export * from './countries'
export * from './continents'
export * from './languages'
export * from './social'
export * from './compare'
export * from './deals'
export * from './marketplace'
export * from './featured'
export * from './nearby'
export * from './recaptcha'
export * from './stripe'
export * from './notifications'
export * from './campaigns'
export * from './contact'
export * from './industry'
export * from './newest'
export * from './questionnaire'
export * from './leagues'
export * from './teams'
export * from './trophies'
