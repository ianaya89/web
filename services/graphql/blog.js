import gql from 'graphql-tag'
import { get } from '../helpers'
import { mockArray } from './_mocks'

export const TagsQuery = {
  mock: mockArray(5),
  query: gql`
    query Tags {
      blog {
        tags {
          id
          name
          occurrences
        }
      }
    }
  `,
  update(data) {
    return get(data, 'blog.tags', [])
  }
}

export const TopPostsQuery = {
  mock: mockArray(4),
  query: gql`
    query TopBlogPosts {
      blog {
        posts(top: true) {
          id
          title
          slug
          title_photo {
            url
          }
          published_at
          read_time
          author
          tags {
            name
          }
        }
      }
    }
  `,
  update(data) {
    return get(data, 'blog.posts', [])
  }
}

export const FilteredPostsQuery = {
  mock: mockArray(7),
  query: gql`
    query FilteredBlogPosts($author: String, $tag: String) {
      blog {
        posts(latest: true, author: $author, tag: $tag) {
          id
          title
          slug
          title_photo {
            url
          }
          published_at
          read_time
          author
          tags {
            name
          }
        }
      }
    }
  `,
  update(data) {
    return get(data, 'blog.posts', [])
  }
}

export const SinglePostQuery = {
  mock: {
    id: 'mock1',
    tags: [],
    title_photo: {},
    related: []
  },
  query: gql`
    query SingleBlogPostWithRelated($postId: ID!) {
      blog {
        post(id: $postId) {
          id
          title
          slug
          excerpt
          title_photo {
            url
          }
          published_at
          read_time
          author
          body
          tags {
            name
          }
          related {
            id
            title
            slug
            title_photo {
              url
            }
            published_at
            read_time
            author
            tags {
              name
            }
          }
        }
      }
    }
  `,

  update(data) {
    return get(data, 'blog.post', { related: [], tags: [] })
  }
}

export const SubmitNewsletter = {
  mutation: gql`
    mutation SubmitNewsletter(
      $email: String!
      $recaptchaToken: String!
      $meta: String
    ) {
      submitNewsletter(
        email: $email
        recaptchaToken: $recaptchaToken
        meta: $meta
      ) {
        id
      }
    }
  `
}
