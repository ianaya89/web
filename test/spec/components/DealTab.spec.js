import { shallowMount } from '@vue/test-utils'
import DealTab from '@/components/deals/DealTab'
import { mockObjects } from '@/services/graphql/_mocks'

describe('DealTab', () => {
  it('renders default animated image', () => {
    const history = mockObjects(4, j => {
      let total = 0
      const items = mockObjects(5, l => {
        const price = 1000
        total += price
        return {
          id: l,
          name: 'Some name',
          price: price,
          amount: 4,
          dimensions: '200 x 40 cm',
          duration: 24,
          photo1: 'some_image_url',
          photo2: 'some_image_url',
          description: 'some description'
        }
      })

      return {
        created_at: '20-12-2012 12:12:12',
        items: items,
        total: total
      }
    })

    const status = 'accepted'
    const props = {
      deal: {
        id: 10,
        created_at: '20-12-2012 12:12:12',
        history: history,
        declined_at: status === 'declined' ? '20-12-2012 12:12:12' : null,
        accepted_at: status === 'accepted' ? '20-12-2012 12:12:12' : null,
        status: status,
        brand: {
          id: 12,
          name: 'some title',
          logo: 'some_image_url',
          representative: {
            name: 'Some name',
            avatar: 'some_image_url'
          }
        }
      },
      active: false
    }

    const wrapper = shallowMount(DealTab, {
      propsData: props
    })
    expect(wrapper.element).toMatchSnapshot()

    wrapper.setProps({
      active: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
