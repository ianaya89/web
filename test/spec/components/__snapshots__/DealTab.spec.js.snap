// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealTab renders default animated image 1`] = `
<div
  class="inactive-tab bg-white w-full px-2 py-3 flex border-b border-r border-grey-l4 cursor-pointer"
>
  <div
    class="w-12 mr-4"
  >
    <avatar-stub
      avatarclasses=""
      class="avatar-active mt-2"
      classes="w-12 h-12"
      rounded="true"
      src=""
      tooltipcolor="light"
      tooltipposition="top"
    />
  </div>
   
  <div
    class="w-3/4 flex flex-col"
  >
    <p
      class="text-h7 lg:text-menu font-medium font-sans leading-tight pt-2"
    >
      
        some title
      
    </p>
     
    <p
      class="text-h7 lg:text-p3 uppercase leading-tight font-sans text-green-d1"
    >
      
      TRANSLATE:app.deals.status.accepted
    
    </p>
     
    <p
      class="text-h7 lg:text-menu text-grey font-medium font-sans leading-tight mt-2"
    >
      
      0 €
    
    </p>
  </div>
</div>
`;

exports[`DealTab renders default animated image 2`] = `
<div
  class="inactive-tab bg-white w-full px-2 py-3 flex border-b border-r border-grey-l4 cursor-pointer"
>
  <div
    class="w-12 mr-4"
  >
    <avatar-stub
      avatarclasses=""
      class="avatar-active mt-2"
      classes="w-12 h-12"
      rounded="true"
      src=""
      tooltipcolor="light"
      tooltipposition="top"
    />
  </div>
   
  <div
    class="w-3/4 flex flex-col"
  >
    <p
      class="text-h7 lg:text-menu font-medium font-sans leading-tight pt-2"
    >
      
        some title
      
    </p>
     
    <p
      class="text-h7 lg:text-p3 uppercase leading-tight font-sans text-green-d1"
    >
      
      TRANSLATE:app.deals.status.accepted
    
    </p>
     
    <p
      class="text-h7 lg:text-menu text-grey font-medium font-sans leading-tight mt-2"
    >
      
      0 €
    
    </p>
  </div>
</div>
`;
